### PR TITLE
Enabled partial month tenancy, made german translations

### PR DESCRIPTION
**Pull request by vectorien**
Hi I have done some work in order to enable the possibility to rent something for some days only, or start renting at a specific date, so the field validation to enter the first of month date is obsolete. I hope this is not against your development goals, but I find it makes sense.
Cheers

### CODE REVIEW

File: main/admin.py [hunk 1]
```diff
@@ -61,12 +61,16 @@ class TenantAdmin(admin.ModelAdmin):
...
-
+    
     def property_link(self, obj):
         url = urlresolvers.reverse(
             'admin:main_property_change', args=(obj.property.id,))
         return format_html(u'<a href={}>{}</a>', mark_safe(url), obj.property)
     property_link.short_description = _('link to the property')
+    
+    class Meta:
+        verbose_name = _("tenant administration")
```
Comment (oadam) [inline main/admin.py]: Aren't you happy with the default name ?

File: main/locale/de/LC_MESSAGES/django.po [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+msgid "reminders"
+msgstr "Erinnerungen"
+
+#: models.py:201
+msgid "start date"
+msgstr "Startdatum"
+
+#: models.py:203
+msgid "end date"
+msgstr "Enddatum"
```
Comment (oadam) [inline main/locale/de/LC_MESSAGES/django.po]: In general, having an uppercase letter at the start of the word is a job for the template [exemple here](https://github.com/oadam/proprio/blob/master/main/templates/main/cashflows.html#L7) . When the untranslated string does contain un uppercase letter, that is an error. Please let's use lowercase everywhere

File: main/migrations/0007_auto_20151115_1833.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
```
Comment (oadam) [inline main/migrations/0007_auto_20151115_1833.py]: I'm not sure why your commit contains so many migration files.
I will check on my machine to try and figure out what's happening

File: main/models.py [hunk 4]
```diff
@@ -119,14 +126,14 @@ def cashflows(self):
         for c in date_sorted:
             balance += c.amount
             result.append(
-                CashflowAndBalance(c.date, c.amount, c.description, balance))
+                CashflowAndBalance(c.date, round(c.amount,2), c.description, round(balance,2)))
```
Comment (oadam) [inline main/models.py]: The rounding should only be made in 1 place (where you perform the partial rent computation)

File: main/locale/de/LC_MESSAGES/django.po (hunk 2 above)
Comment (vectorien) [inline main/locale/de/LC_MESSAGES/django.po]: ok, true

File: main/models.py [hunk 5]
```diff
@@ -252,9 +262,9 @@ def payments_to_cashflows(date, payments):
         if p.date > date:
             continue
         if p.description:
-            d = _('payment "%(description)s"') % {'description': p.description}
+            d = _('PAYMENT "%(description)s"') % {'description': p.description}
```
Comment (oadam) [inline main/models.py]: I think lowercase is fine here ?

File: main/models.py [hunk 6]
```diff
@@ -263,28 +273,66 @@ def fees_to_cashflows(date, fees):
             for x in fees if x.date <= date]
 
 
-def revision_to_cashflows(rev, end_date):
+def revision_to_cashflows(rev, end_date, tenancy_begin_date = date(2015, 6, 1), tenancy_end_date = date(2015, 12, 13)):
```
Comment (oadam) [inline main/models.py]: could you use the revision `start_date` and `end_date` instead of additionnal parameters ? `tenancy_begin_date` and `tenancy_end_date` are not used in the computations. I introduced them for historical reasons (tenants that were here before I created the system)

Comment (oadam) [inline main/models.py]: Also could you add a test case please ?

File: main/templates/base.html [hunk 7]
```diff
@@ -1,5 +1,6 @@
 {% load lineage %}
 {% load i18n %}
+{% load l10n %}
```
Comment (oadam) [inline main/templates/base.html]: Why did you add this ?

File: main/templates/main/tenants.html [hunk 8]
```diff
@@ -32,7 +33,7 @@
 				<span class="trend" normal-min="{{ t.normal_min }}"> {{ t.trend }}</span>
 			</td>
 			<td>
-				<button type="button" class="btn btn-default receipts"><span class="glyphicon glyphicon-envelope"></span> {% trans "write letter"|title %}</button>
```
Comment (oadam) [inline main/templates/main/tenants.html]: You can remove it

File: proprio/settings.py [hunk 9]
```diff
@@ -18,7 +18,7 @@
 # See https://docs.djangoproject.com/en/1.6/howto/deployment/checklist/
 
 # SECURITY WARNING: keep the secret key used in production secret!
-SECRET_KEY = 'ov3sfsyj4*@0)m@3-j+m%%(0ms*et@d0hb+pl#6z2m(t=be*(n'
```
Comment (oadam) [inline proprio/settings.py]: Can you revert this part ?

File: proprio/settings.py [hunk 10]
```diff
@@ -90,9 +90,9 @@
 # Internationalization
 # https://docs.djangoproject.com/en/1.6/topics/i18n/
 
-LANGUAGE_CODE = 'en-us'
```
Comment (oadam) [inline proprio/settings.py]: You can delete `LANGUAGE_CODE` and `TIME_ZONE`

**Comment by oadam** [comment]
Hi,

I made a couple of remarks. The most important one is that I would prefer that we don't use `tenancy.start_date` and `tenancy.end_date` to generates rents. Instead I would prefer to use `start_date` and `end_date` from `RentRevision`.

Also next time can you try to split your commit in smaller pull requests ?

Apart from that the change looks good, I'm looking forward to merging it.

Thanks,
Olivier

File: main/models.py (hunk 5 above)
Comment (vectorien) [inline main/models.py]: hm i thought it stands out better if it is all uppercase..

Comment (oadam) [inline main/models.py]: Ok. We can keep uppercase

File: main/models.py (hunk 6 above)
Comment (vectorien) [inline main/models.py]: Hi, in my understanding, in order to compute partial rents, we do need those params. Reason: A rent revision can differ from a tenancy period..

test case: as soon as I find time. I know, I know, tests should come first ...

File: main/templates/base.html (hunk 7 above)
Comment (vectorien) [inline main/templates/base.html]: uhm I had some problems, that are now obsolete, so we can delete this

File: main/migrations/0007_auto_20151115_1833.py (hunk 3 above)
Comment (vectorien) [inline main/migrations/0007_auto_20151115_1833.py]: well I did a couple of migrations in the process...

File: main/migrations/0010_auto_20151118_0825.py [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('main', '0009_auto_20151116_1022'),
+    ]
+
+    operations = [
+        migrations.CreateModel(
```
Comment (oadam) [inline main/migrations/0010_auto_20151118_0825.py]: For example this line is super suspicious, as the `building` model already exists...

File: proprio/settings.py (hunk 10 above)
Comment (vectorien) [inline proprio/settings.py]: done

File: main/admin.py (hunk 1 above)
Comment (vectorien) [inline main/admin.py]: I found it clearer to fully write it

File: main/migrations/0010_auto_20151118_0825.py (hunk 11 above)
Comment (vectorien) [inline main/migrations/0010_auto_20151118_0825.py]: as said above, I had disable the property classes for a while

File: proprio/settings.py (hunk 10 above)
Comment (vectorien) [inline proprio/settings.py]: done

File: main/templates/main/tenants.html (hunk 8 above)
Comment (vectorien) [inline main/templates/main/tenants.html]: I thought i good to have the property info also in the main panel, to have a good overview. Later, to be able to model reality, I think it would be a good idea to separate the tenant class into two classes: 1.  tenant (the person) and 2. tenancy (the duration of a rental)

File: main/locale/de/LC_MESSAGES/django.po (hunk 2 above)
Comment (oadam) [inline main/locale/de/LC_MESSAGES/django.po]: Ok I did not know German had a special handling of uppercase. You can definitely keep uppercase translation !

File: main/migrations/0007_auto_20151115_1833.py (hunk 3 above)
Comment (oadam) [inline main/migrations/0007_auto_20151115_1833.py]: But if I merge as is, I think will loose my property data in the first migration and then an empty data will be recreated.
Maybe the simplest approach is to squash your commits into a single commit containing 0 or 1 migration file ?

Comment (vectorien) [inline main/migrations/0007_auto_20151115_1833.py]: sure. Ill just delete my migration files..

**Comment by vectorien** [comment]
yeah sorry. at the time I thought about a pr, it was already so big
 next time will be smaller. Maybe for example the test case in a later pr.

As for the mentioned variables, I commented further up

**Comment by oadam** [comment]
Hi @vectorien,
I've cherry-picked some of your suggestions to the main branch. I did not have time to fully review the "meat" of the change (partial rents) due to lack of time but I've noticed that it broke some of the automated tests.
I've reconfigured the repository so that it uses travis-ci, so hopefully broken tests will stand out more in future pull requests.
I think the change could be smaller (and do not break tests :smile: ). My recommended approach is to : 
- remove the `validate_month` validator
- add a test case in `test_revision_to_fees` that uses a partial month
- slightly change the code in `main/main.py` to make the test pass
If you have any question, please feel free to chat me during the day on hangout or skype (skype id: adam_olivier)
Cheers !

**Comment by vectorien** [comment]
Ok then lets close this pr and I implement the things u mentioned , then i'll make a new pr. cool?

**Comment by oadam** [comment]
Sure.
Thanks !
